src/static/js/event_animals.js

```diff
@@ -0,0 +1,319 @@
+            if (!common.has_permission("cea")) { $("#button-addanimals").hide(); }
+            if (!common.has_permission("cea")) { $("#button-removeanimal").hide(); }
+            if (!common.has_permission("cea")) { $("#button-animalarrived").hide(); }
+            if (!common.has_permission("cea")) { $("#button-animalendactivefoster").hide(); }
+
+        },
+
+        validation: function(){
+            header.hide_error();
+            validate.reset();
+            if(common.trim($("#startdate").val()) == ""){
+                header.show_error(_("Event must have a start date."));
+                validate.highlight("startdate");
+                validate.dirty(false);
+                return false;
+            }
+            if (common.trim($("#enddate").val()) == ""){
+                header.show_error(_("Event must have an end date."));
+                validate.highlight("enddate");
+                validate.dirty(false);
+                return false;
+            }
+            if (common.trim($("#address").val()) == ""){
+                header.show_error(_("Event must have an address."));
+                validate.highlight("address");
+                validate.dirty(false);
+                return false;
+            }
+            // mandatory additional fields
+            if (!additional.validate_mandatory()) { return false; }
+            return true;
+        },
+
+        sync: function(){
+
+            // Load the data into the controls for the screen
+            if (common.querystring_param("filter")) {
+                $("#filter").select("value", common.querystring_param("filter"));
+            }
+
+            // Update on-screen fields from the data and display the screen
+            event_animals.enable_widgets();
+
+            // Dirty handling
+            validate.bind_dirty([ "eventanimal_" ]);
+        },
+
+        destroy: function() {
+            common.widget_destroy("#dialog-arrived");
+            common.widget_destroy("#dialog-endactivefoster");
+            common.widget_destroy("#dialog-addanimal");
+            common.widget_destroy("#dialog-addanimals");
+            tableform.dialog_destroy();
+        },
+
+        name: "event_animals",
+        animation: "formtab",
+        autofocus: "#eventtype",
+        title: function() {
+            var e = controller.event;
+            var dates_range = "";
+            if(format.date(e.STARTDATETIME) == format.date(e.ENDDATETIME))
+                    dates_range = format.date(e.STARTDATETIME);
+                else
+                    dates_range = format.date(e.STARTDATETIME) + " - " + format.date(e.ENDDATETIME);
+            return dates_range + " " + e.EVENTNAME + " " + [e.EVENTADDRESS, e.EVENTTOWN, e.EVENTCOUNTY, e.EVENTCOUNTRY].filter(Boolean).join(", ");
+        },
+
+        routes: {
+            "event_animals": function() { common.module_loadandstart("event_animals", "event_animals?id=" + this.qs.id); }
```

Comment:
I think this is your issue for offset not working - your only route for this module assumes a single id parameter. Change this to:

common.module_loadandstart("event_animals", "event_animals?" + this.rawqs);

You'll be able to pass any parameters then.